src/Core/Security/Application/Repository/ReadAccessGroupRepositoryInterface.php

```diff
@@ -28,11 +28,26 @@
 
 interface ReadAccessGroupRepositoryInterface
 {
+    /**
+     * Find all access groups
+     *
+     * @return AccessGroup[]
+     */
+    public function findAll(): array;
+
     /**
      * Find all access groups according to a contact.
      *
      * @param ContactInterface $contact Contact for which we want to find the access groups.
      * @return AccessGroup[]
      */
     public function findByContact(ContactInterface $contact): array;
+
+    /**
+     * Find all access groups according to a contact with filter
+     *
+     * @param int $userId
+     * @return AccessGroup[]
+     */
+    public function findByContactWithFilter(int $userId): array;
```

Comment:
`WithFilter` means using query search parameter ?
if yes, it's not consistent with `findAll` which also use query search parameter